lib/neo4j/migrations/migration_file.rb

```diff
@@ -0,0 +1,24 @@
+module Neo4j
+  module Migrations
+    class MigrationFile
+      attr_reader :file_name, :symbol_name, :class_name, :version
+
+      def initialize(file_name)
+        @file_name = file_name
+        extract_data!
+      end
+
+      def create
+        require file_name
+        class_name.constantize.new(version)
+      end
+
+      private
+
+      def extract_data!
+        @version, @symbol_name = File.basename(file_name, '.rb').split('_', 2)
+        @class_name = @symbol_name.split('_').map(&:capitalize).join('')
```

Comment:
Yup!  We use the modules `PerThreadRegistry`, `Notifications`, `Inflector`, `Concern`, and a few core extensions, so it's pretty ingrained into the gem by now ;)  But we try to require only the sections that we need